### ISSUE TITLE
"Numeric" comparison with other objects

### ISSUE THREAD

**Issue by CollinAlpert** [TrigerSoft/jaque#37]
I am trying to implement comparisons like "greater than" or "less than" for non-numeric Java objects. For example a LocalDate. I want the LocalDate#isAfter(LocalDate) method to be evaluated as a "greater than" operation. What I tried (simplified version), was this:
```java
LocalDate date1 = LocalDate.of(2016, 8, 23);
LocalDate date2 = LocalDate.now();
Expression.binary(ExpressionType.GreaterThan, Expression.constant(date1), Expression.constant(date2));
```
What I found was that JaQue throws an `IllegalArgumentException`in the `createNumericComparison` method, because `java.time.LocalDate` is not a numeric value.
Any way I can get this to work?

**Comment by kostat** [comment]
Strictly speaking this wouldn't be Java. But I can consider extend it to
objects implementing Comparable interface

**Comment by CollinAlpert** [comment]
That would be great!

**Comment by kostat** [comment]
Ok, will do

**Comment by kostat** [comment]
released in 2.4.3 version

**Comment by CollinAlpert** [comment]
Thanks so much!
